src/pages/index.js

```diff
@@ -5,28 +5,79 @@ import Section from "../components/Section.js";
 import {
   initialCards,
   selectors,
-  addCardImgUrlInput,
-  addCardTitleInput,
+  imageProfileAvatar,
   nameInput,
   jobInput,
   ImagePopupSelector,
   addCardButton,
   config,
   addCardFormElement,
+  editAvatarFormElement,
   editProfileFormElement,
   editProfileButton,
 } from "../utils/Constants.js";
 import Card from "../components/Card.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
+import { Api } from "../components/Api.js";
+import DeleteConfirmationPopup from "../components/DeleteConfirmationPopup.js";
 
 const handleImageClick = (data) => {
   imagePopup.open(data);
 };
 
 const createCard = (cardData) => {
-  const card = new Card(cardData, selectors.Cardtemplate, handleImageClick);
+  const card = new Card(
+    cardData,
+    selectors.Cardtemplate,
+    handleImageClick,
+    function handleDeleteModal() {
+      deleteCardForm.open();
+
+      deleteCardForm.setSubmitAction(() => {
+        deleteCardForm.renderLoading(true, "Deleting...");
+        api
+          .deleteCard(card.id)
+          .then((res) => {
+            console.log("The delete card function is being called");
+            // delete the card element
+            card.deleteCard();
+
+            deleteCardForm.close();
+          })
+          .finally(() => {
+            deleteCardForm.renderLoading(true, "Yes");
```

Comment:
`.renderLoading(true` 


`true` is for setting the **loading** effect, but you need to use `false` in finally to set the default button text

Reply:
I use false as the value in the method now